### ISSUE TITLE
[Serve] Version Support in 2.X API

### ISSUE THREAD

**Issue by sihanwang41** [ray-project/ray#31928]
### Description

Add a version support in the serve.run() API to make sure the deployment will not be redeployed with same version.

```
server.run(xxxx, version: str = None)
```

When the version is provided, serve will set all deployments to the version, and the same version of the deployment won't be redeployed.

(Also need to be supported in the dashboard serve agent for REST API deploy)

**Comment by zoltan-fedor** [comment]
Just to clarify, this is to restore the functionality of the `version` tag in deployments when using API v1.x.

With API v1.x when you deploy a Serve deployment with a version specified (by calling the `deploy()` method), if such a deployment with such name and version is already running, then it will just return the ref for that. 
If such a deployment (name and version matching) does not exist, then it will create (or redeploy) the deployment and then return its ref. 

This is basically the well-known `get_or_create` pattern of `classmethods` of OOP.

Again, this existed in API v1.x - and being used by people. We are trying to restore existing functionality into API v2.x.

**Comment by sihanwang41** [comment]
Yes, @zoltan-fedor. We are not going to deprecate the version functionality in v1.x API until v2.x API have supports with it.

> If such a deployment (name and version matching) does not exist, then it will create (or redeploy) the deployment and then return its ref.
That is the correct expected behavior for v2.x API with this support.

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
